src/Archives/ClmFile.cpp

```diff
@@ -273,8 +273,13 @@ namespace Archives
 		uint32_t offset = headerSize + internalNames.size() * sizeof(IndexEntry);
 		for (std::size_t i = 0; i < internalNames.size(); ++i)
 		{
+#pragma warning( push )
+#pragma warning( disable : 4996 ) // Disable MSVC warning 'strncpy': This function or variable may be unsafe. Consider using strncpy_s instead.
```

Comment:
Looks correct.

This is slightly wordier than I expected it to be. I had erroneously thought the push and disable could be combined. Turns out there is an even better option though. Checking MSDN for [warning](https://msdn.microsoft.com/en-us/library/2c8f766e.aspx), I noticed there is a `suppress` option, which will push the current state, disable a warning for the next line, and then pop the state.